components/places/src/sync/history/store.rs

```diff
@@ -0,0 +1,184 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::result;
+use std::cell::Cell;
+use std::ops::Deref;
+use error::*;
+use failure;
+use rusqlite::{Connection};
+use rusqlite::{types::{ToSql, FromSql}};
+use serde_json;
+use sql_support::{ConnExt};
+use sync15_adapter::{
+    GlobalState,
+    GlobalStateProvider,
+    ClientInfo,
+    IncomingChangeset,
+    OutgoingChangeset,
+    Store,
+    ServerTimestamp
+};
+use sync15_adapter::request::{CollectionRequest};
+
+use super::record::{HistorySyncRecord};
+use super::incoming_plan::{IncomingPlan, plan_incoming_record};
+use super::super::super::storage::history_sync::{apply_synced_visits} ;
+
+
+static LAST_SYNC_META_KEY:    &'static str = "history_last_sync_time";
+static GLOBAL_STATE_META_KEY: &'static str = "history_global_state";
+
+// Lifetime here seems wrong
```

Comment:
We've discussed this -- I think we probably want ownership (possibly of a PlacesDb) here. That or it should stay shortlived, which isn't great given `ClientInfo`, but also doesn't matter *that* much in the short term.